test/FunctionalTests/Tests/DataServices/UnitTests/ClientCSharpUnitTests/DataWebClientCSharp/LinqTests.cs

```diff
@@ -8153,6 +8153,34 @@ public void QueryRowCount_UriBuilder()
             }
         }
 
+        [TestMethod]
+        public void VerifyCountIsNotCarriedOver()
+        {
+            string expectedLongCountFunctionUri = "/service.svc/Leagues/LongCountFunction(Param1='Param1',Param2='Param2')?$orderby=ID";
+            string expectedCountFunctionUri = "/service.svc/Leagues/CountFunction(Param1='Param1',Param2='Param2')?$orderby=ID";
+
+            var countQuery = context.CreateQuery<League>("Leagues");
+            var longcountQuery = context.CreateQuery<League>("Leagues");
+
+            var countFunctionQuery = countQuery.CreateFunctionQuery<League>("CountFunction", true, new UriOperationParameter("Param1", "Param1"),
+                    new UriOperationParameter("Param2", "Param2"));
+            var longCountFunctionQuery = longcountQuery.CreateFunctionQuery<League>("LongCountFunction", true, new UriOperationParameter("Param1", "Param1"),
+                    new UriOperationParameter("Param2", "Param2"));
+            try
+            {
+                //check for both count an longcount
+                countFunctionQuery.Count();
+                longCountFunctionQuery.LongCount();
+
+            }
+            catch(DataServiceQueryException)
+            {
+                var countOrderQuery = countFunctionQuery.OrderBy(x => x.ID);
+                var longCountOrderQuery = longCountFunctionQuery.OrderBy(x => x.ID);
+                Assert.IsTrue(countOrderQuery.ToString().EndsWith(expectedCountFunctionUri), "Wrong uri generated with count option");
+                Assert.IsTrue(longCountOrderQuery.ToString().EndsWith(expectedLongCountFunctionUri), "Wrong uri generated with longcount option");

```

Comment:
Why do a .EndsWith instead of a .Contains?

Reply:
The Query contains the full URL including the host. We are not checking the host. Remember the port of the inmemory service changes on every service instance.